components/base/kinematicbase/kinematics.go

```diff
@@ -103,6 +115,14 @@ func (ddk *differentialDriveKinematics) GoToInputs(ctx context.Context, desired
 			return err
 		}
 
+		col, err := validRegion.CollidesWith(spatialmath.NewPoint(r3.Vector{X: current[0].Value, Y: current[1].Value}, ""))
+		if err != nil {
+			return err
+		}
+		if !col {
+			return errors.New("base has deviated too far from path")
+		}
+
 		// get to the x, y location first - note that from the base's perspective +y is forward
 		desiredHeading := math.Atan2(current[1].Value-desired[1].Value, current[0].Value-desired[0].Value)
 		if commanded, err := ddk.issueCommand(ctx, current, []referenceframe.Input{desired[0], desired[1], {desiredHeading}}); err == nil {
```

Comment:
I'm just now noticing that this error isn't ever checked 🙃

Reply:
isn't it checked to make sure it's nil at the end of the if. do we also want to throw error if err not nil